src/main/kotlin/org/openmbee/flexo/mms/routes/sparql/ScratchUpdate.kt

```diff
@@ -0,0 +1,91 @@
+package org.openmbee.flexo.mms.routes.sparql;
+
+import io.ktor.http.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import org.apache.jena.sparql.modify.request.*
+import org.apache.jena.update.UpdateFactory
+import org.openmbee.flexo.mms.*
+import org.openmbee.flexo.mms.routes.SCRATCHES_PATH
+import org.openmbee.flexo.mms.server.sparqlUpdate
+
+/**
+ * User submitted SPARQL Update to a scratch space
+ */
+fun Route.updateScratch() {
+    sparqlUpdate("$SCRATCHES_PATH/{scratchId}/update") {
+        parsePathParams {
+            org()
+            repo()
+            scratch()
+            branch()
+        }
+
+        // construct the scratch's named graph IRI
+        val scratchGraph = "mor-graph:Scratch.$scratchId"
+
+        // parse query
+        val sparqlUpdateAst = try {
+            UpdateFactory.create(requestContext.update)
+        } catch (parse: Exception) {
+            throw UpdateSyntaxException(parse)
+        }
+
+        val operations = sparqlUpdateAst.operations
+
+        assertOperationsAllowed(operations)
+
+        // merge the client prefixes with internal ones
+        val mergedPrefixMap = HashMap(sparqlUpdateAst.prefixMapping.nsPrefixMap)
```

Comment:
there shouldn't be a need to merge the mms ones since there's nothing we're doing with branches/commits/transactions